projects/wave-core-new/src/lib/layers/layer.model.ts

```diff
@@ -30,72 +35,112 @@ export abstract class Layer implements ToDict<LayerDict> {
     }
 
     protected constructor(config: {
+        id?: number,
         name: string,
         workflowId: string,
+        isVisible: boolean,
+        isLegendVisible: boolean,
         symbology: AbstractSymbology,
     }) {
-        this._name = config.name;
+        this.id = config.id ?? Layer.nextLayerId++;
+
+        this.name = config.name;
         this.workflowId = config.workflowId;
+        this.isVisible = config.isVisible;
+        this.isLegendVisible = config.isLegendVisible;
         this.symbology = config.symbology;
     }
 
-    get name(): string {
-        return this._name;
-    }
-
-    get isVisible(): boolean {
-        return this._isVisible;
-    }
-
-    get isLegendVisible(): boolean {
-        return this._isLegendVisible;
-    }
+    // TODO: remove method, here?
+    abstract updateFields(changes: {
```

Comment:
could also be a trait/interface

Reply:
It is just the question if we want to allow using the update method if you don't know the layer type.